app/connector/kafka/src/main/java/io/syndesis/connector/kafka/KafkaMetaDataRetrieval.java

```diff
@@ -71,46 +87,64 @@ protected SyndesisMetadata adapt(CamelContext context, String componentId, Strin
     }
 
     /**
-     * Query the strimzi brokers available on this kubernetes environment to suggest auto discovered urls.
+     * Query the strimzi brokers available on this kubernetes environment
+     * to suggest auto discovered urls.
      */
     @Override
-    @SuppressWarnings("unchecked")
     public SyndesisMetadataProperties fetchProperties(CamelContext context, String componentId,
                                                       Map<String, Object> properties) {
         List<PropertyPair> brokers = new ArrayList<>();
         try (KubernetesClient client = new DefaultKubernetesClient()) {
-
-            //Filter by strimzi resources
-            final CustomResourceDefinitionContext build = new CustomResourceDefinitionContext.Builder()
-                                                              .withGroup("kafka.strimzi.io")
-                                                              .withName("kafkas.kafka.strimzi.io")
-                                                              .withPlural("kafkas")
-                                                              .withScope("Namespaced")
-                                                              .withVersion("v1beta1")
-                                                              .build();
-
-            final List<Map<String, ?>> items = (List<Map<String, ?>>) client.customResource(build).list().get("items");
-
-            for (Map<String, ?> item : items) {
-                //Extract an identifier of this broker
-                final Map<?, ?> metadata = (Map<?, ?>) item.get("metadata");
-                String id = metadata.get("namespace") + "::" + metadata.get("name");
-
-                //Add the list of addresses for this item
-                Map<?, ?> status = (Map<?, ?>) item.get("status");
-                List<Map<String,
-                            List<Map<String, Object>>>> listeners =
-                    (List<Map<String, List<Map<String, Object>>>>) status.get("listeners");
-                listeners.stream().filter(typesAllowed).forEach(
-                    listener -> getAddress(listener, brokers, id));
-            }
+            client.customResourceDefinitions().list().getItems()
+                .stream().filter(isKafkaBroker)
+                .forEach(kafka -> processKafkaCRD(brokers, client, kafka));
+        } catch (Throwable t) {
+            LOG.warn("Couldn't auto discover any broker.", t);
```

Comment:
I'd do something like this, so that we don't log stack traces if debug log level is not enabled.

```suggestion
            LOG.warn("Couldn't auto discover any broker.");
            LOG.debug("Couldn't auto discover any broker.", t);
```